Bio/SeqUtils/ProtParam.py

```diff
@@ -161,25 +161,32 @@ def flexibility(self):
         a window=9. The parameters used are optimized for determining the
         flexibility.
         """
-        flexibilities = ProtParamData.Flex
+        bnorm = ProtParamData.Flex
+        bnorm_0 = ProtParamData.Flex_0
+        bnorm_1 = ProtParamData.Flex_1
+        bnorm_2 = ProtParamData.Flex_2
         window_size = 9
+        bnorm_nc = [0] * (self.length - 2)
         weights = [0.25, 0.4375, 0.625, 0.8125, 1]
         scores = []
 
-        for i in range(self.length - window_size):
-            subsequence = self.sequence[i : i + window_size]
-            score = 0.0
+        # Calculate neighbor-correlated B_norm values first
+        for i, res in enumerate(self.sequence[1 : self.length - 1]):
+            bnorm_nc[i] = bnorm_0[res]
+            if bnorm[self.sequence[i]] < 1 or bnorm[self.sequence[i + 2]] < 1:
+                bnorm_nc[i] = bnorm_1[res]
+            if bnorm[self.sequence[i]] < 1 and bnorm[self.sequence[i + 2]] < 1:
+                bnorm_nc[i] = bnorm_2[res]
```

Comment:
If I am reading this right, what you have here is:

```python
if A or B:
   bnorm_nc[i] = bnorm_1[res]
if A and B:
   bnorm_nc[i] = bnorm_2[res]
```

This seems inefficient when both A and B are true, since the first if-statement and code do not need to be run. How about:

```python
if A and B:
   bnorm_nc[i] = bnorm_2[res]
elif A or B:
   bnorm_nc[i] = bnorm_1[res]
```

Reply:
Yeah, these lines are not the most efficient. I was just following the original implementation. I will update it.